### PR TITLE
feat: Create group optionally

### PR DESCRIPTION
**Pull request by antonbabenko**
Closes #216 (no access to fork by `yo61`)

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -2,18 +2,14 @@
...
-  )[0]
+  this_sg_id = var.create_sg ? concat(aws_security_group.this.*.id, aws_security_group.this_name_prefix.*.id, [""])[0] : var.security_group_id
 }
 
 ##########################
 # Security group with name
 ##########################
 resource "aws_security_group" "this" {
-  count = var.create && false == var.use_name_prefix ? 1 : 0
+  count = var.create && var.create_sg && !var.use_name_prefix ? 1 : 0
```
Comment (robinbowes) [inline main.tf:12]: Are you OK with this change, ie. `!var.use_name_prefix` instead of `false == var.use_name_prefix` ?

File: main.tf [hunk 2]
```diff
@@ -2,18 +2,14 @@
 # Get ID of created Security Group
 ##################################
 locals {
-  this_sg_id = concat(
-    aws_security_group.this.*.id,
-    aws_security_group.this_name_prefix.*.id,
-    [""],
-  )[0]
+  this_sg_id = var.create_sg ? concat(aws_security_group.this.*.id, aws_security_group.this_name_prefix.*.id, [""])[0] : var.security_group_id
```
Comment (antonbabenko) [inline main.tf:5]: This is necessary because the `disabled` example was not working with just `coalesce()`.

**Review by robinbowes — APPROVED** [review]
Added a comment, but LGTM.

**Comment by antonbabenko** [comment]
[v4.2.0](https://github.com/terraform-aws-modules/terraform-aws-security-group/releases/v4.2.0) has been just released.

File: main.tf (hunk 1 above)
Comment (antonbabenko) [reply main.tf:12]: Yes, looks good to me.

**Comment by antonbabenko** [comment]
Thank you @robinbowes for pushing this module today :)

**Comment by robinbowes** [comment]
Thanks for merging - 965 days after my initial PR 😀

File: variables.tf [hunk 3]
```diff
@@ -7,14 +7,28 @@ variable "create" {
   default     = true
 }
 
+variable "create_sg" {
```
Comment (robinbowes) [inline variables.tf:10]: I didn't notice before, but you changed the parameter name - it was `create_group`, you changed to `create_sg`. Was there a reason to do that?

File: variables.tf [hunk 4]
```diff
@@ -7,14 +7,28 @@ variable "create" {
...
+
 variable "vpc_id" {
   description = "ID of the VPC where to create security group"
   type        = string
+  default     = null
 }
 
 variable "name" {
-  description = "Name of security group"
+  description = "Name of security group - not required if create_group is false"
```
Comment (robinbowes) [inline variables.tf:29]: Original parameter name was `create_group` - now changed to `create_sg`

File: variables.tf (hunk 3 above)
Comment (antonbabenko) [reply variables.tf:10]: To be more like the rest of terraform-aws-modules, also "group" is a bit unclear (db_subnet_group in rds/vpc, origin_group in cloudfront).

**Pull request merged**